run_mount.go

```diff
@@ -69,12 +77,59 @@ func generateMountOpts(clicontext *cli.Context, imageVolumes map[string]struct{}
 			continue
 		}
 		anonVolName := idgen.GenerateID()
+
 		logrus.Debugf("creating anonymous volume %q, for \"VOLUME %s\"",
 			anonVolName, imgVolRaw)
 		anonVol, err := volStore.Create(anonVolName)
 		if err != nil {
 			return nil, nil, err
 		}
+
+		client, ctx, cancel, err := newClient(clicontext)
+		if err != nil {
+			return nil, nil, err
+		}
+		defer cancel()
+
+		img, err := client.ImageService().Get(ctx, clicontext.Args().First())
+		if err != nil {
+			return nil, nil, err
+		}
+
+		i := containerd.NewImage(client, img)
+
+		if err := i.Unpack(ctx, clicontext.String("snapshotter")); err != nil {
+			return nil, nil, errors.Wrap(err, "error unpacking image")
+		}
+
+		diffIDs, err := i.RootFS(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		chainID := identity.ChainID(diffIDs).String()
+
+		s := client.SnapshotService(clicontext.String("snapshotter"))
+
+		tempDir, err := ioutil.TempDir(os.Getenv("GOTMPDIR"), "initialC")
+		if err != nil {
+			return nil, nil, err
+		}
+		defer os.RemoveAll(tempDir)
+
+		var mounts []mount.Mount
+		mounts, err = s.View(ctx, tempDir, chainID)
+
+		if err := mount.All(mounts, tempDir); err != nil {
+			if err := s.Remove(ctx, tempDir); err != nil && !errdefs.IsNotFound(err) {
+				return nil, nil, err
+			}
+			return nil, nil, err
+		}
+		target := strings.Join([]string{tempDir, imgVol}, "")
+
+		//copying up initial contents of the mount point directory
+		copyExistingContents(target, anonVol.Mountpoint)
```

Comment:
Named volumes need this as well

Reply:
@AkihiroSuda  I made modification. FYI : in docker named volume do not copy initial content ( that may override source volume)

Reply:
@AkihiroSuda sorry I mixed mount bind and named volume. I added `copyExistingContents`to named volume